src/genome.cpp

```diff
@@ -57,3 +71,28 @@ Gene *Genome::getGene(int index) {
 int Genome::getSize() {
 	return genes.size();
 }
+
+//The random genes will need to be added to everytime we make a new gene type. We might want to come up with a better solution in the future.
+Gene *Genome::GenerateRandomGene() {
+	RandomNumberGenerator rand = RandomNumberGenerator();
+	int generatedValue = rand.randf_range(0,100);
+
+	if(generatedValue >= CELL_MEMBRANE_CHANCE[0] && generatedValue < CELL_MEMBRANE_CHANCE[1])
+		return new CellMembraneGene();
+
+	if(generatedValue >= FLAGELLA_CHANCE[0] && generatedValue < FLAGELLA_CHANCE[1])
+		return new FlagellaGene();
+
+	if(generatedValue >= MITOCHONDRIA_CHANCE[0] && generatedValue < MITOCHONDRIA_CHANCE[1])
+		return new MitochondriaGene();
+
+	if(generatedValue >= NUCLEUS_CHANCE[0] && generatedValue < NUCLEUS_CHANCE[1])
+		return new NucleusGene();
+
+	if(generatedValue >= RIBOSOME_CHANCE[0] && generatedValue < RIBOSOME_CHANCE[1])
+		return new RibosomesGene();
+
+	if(generatedValue >= MODIFIER_CHANCE[0] && generatedValue < MODIFIER_CHANCE[1])
+		return randomModifierGene();
+	return randomModifierGene();
```

Comment:
If the default is to return a random modifier gene, checking for the modifier chance is redundant. It may not make sense to return a random gene if it somehow gets to that point. I think randf_range is inclusive though, so if it actually generated exactly 100, currently it would actually get to that point.

Reply:
Yeah, I agree we don't really need to check for modifier gene. I think I meant to set it to be less than or equal to. We can just get rid of the check